core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -794,6 +797,45 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
   }
 
   /**
+   * Reads in a directory of Avro files from HDFS, a local file system (available on all nodes), or
+   * any Hadoop-supported file system URI. The records are read in as Generic Avro records. This
+   * also allows a user to register a schema with Kryo, if they so choose to.
+   *
+   * You can do the following if you know the schema ahead of time:
+   * {{{
+   *   val schema = new Schema.Parser().parse(schemaString)
+   *   sc.avroFile("/input-path", schema)
+   * }}}
+   *
+   * or just:
+   * {{{
+   *   sc.avroFile("/input-path")
+   * }}}
+   */
+  def avroFile(path: String, schemas: Schema*): RDD[GenericRecord] = {
```

Comment:
Hm, so this API allows to read Avro records that have completely unrelated schemas as `GenericRecord`s?  Does this mean users may get Avro `GenericRecord`s in different shape?  This seems to be a little bit unintuitive to me.  Is this a common case when dealing with Avro files?  Especially, is this related to schema merging?  Maybe we should restrict this PR to only support reading a single type of Avro records at a time (namely, remove the `*`).  We can always provide more powerful APIs later if they are proved to be practical.